apollo-federation/src/merger/merge_input.rs

```diff
@@ -0,0 +1,605 @@
+                        missing_subgraphs_str
+                    ),
+                });
+                } else {
+                    let mut present_subgraphs = Vec::new();
+                    let mut locations = Vec::new();
+
+                    // Extract nodes and create locations for fields that exist
+                    for (idx, field) in subgraph_fields.iter() {
+                        if let Some(field_component) = field {
+                            present_subgraphs.push(&self.names[*idx]);
+
+                            let node = &field_component.node;
+
+                            // Create locations if we have subgraph information
+                            if let Some(subgraph) = self.subgraphs.get(*idx) {
+                                let field_locations =
+                                    subgraph.schema().node_locations(node).map(|loc| {
+                                        SubgraphLocation {
+                                            subgraph: subgraph.name.clone(),
+                                            range: loc,
+                                        }
+                                    });
+                                locations.extend(field_locations);
+                            }
+                        }
+                    }
+
+                    let present_subgraphs_str = present_subgraphs
+                        .iter()
+                        .map(|s| s.as_str())
+                        .collect::<Vec<_>>()
+                        .join(", ");
+                    let missing_subgraphs_str = missing_subgraphs
+                        .iter()
+                        .map(|s| s.as_str())
+                        .collect::<Vec<_>>()
+                        .join(", ");
+
+                    self.error_reporter.add_hint(CompositionHint {
+                        code: HintCode::InconsistentInputObjectField.code().to_string(),
+                        message: format!(
+                            "Input object field \"{}\" will not be added to \"{}\" in the supergraph as it does not appear in all subgraphs: it is defined in {} but not in {}",
+                            dest_field.field_name,
+                            dest.type_name,
+                            present_subgraphs_str,
+                            missing_subgraphs_str
+                        ),
+                        locations,
+                    });
+                }
+                // Note that we remove the element after the hint/error because we access the parent in the hint message.
+                dest_field.remove(&mut self.merged)?;
+            }
+        }
+
+        // We could be left with an input type with no fields, and that's invalid in GraphQL
+        let final_input_object = dest.get(self.merged.schema())?;
+        if final_input_object.fields.is_empty() {
+            self.error_reporter.add_error(CompositionError::InternalError {
```

Comment:
I believe this should be a specific `EMPTY_MERGED_INPUT_TYPE` composition error that should include location information (`nodes` in JS)